### DISCUSSION TITLE
Signature Spoofing needs to be made non mandatory since it messes with history, closed captions and maybe more.

### DISCUSSION THREAD

**Discussion by AgelosL** [ReVanced/revanced-discussions#941] [General]
Unless all the bugs are ironed out, I think it's best if the patch was made non mandatory when patching microg. I know there is a problem with users in the UK but literally the rest of the world shouldn't have to deal with new problems just because the signature spoofing buff isn't ready. No-one should or will forbid you from using the signature spoofing patch if you have playback problem bc u are in the UK but this shouldn't be forced.

**Comment by oSumAtrIX** [comment]
In that case please PR to remove the dependency of microg-support to that patch by removing the @DependsOn annotation from the patch class file. That way, the patch will be able to be excluded even if microg-support is included.

**Reply by AgelosL** [reply]
https://github.com/revanced/revanced-patches/blob/main/src/main/kotlin/app/revanced/patches/youtube/misc/fix/playback/patch/SpoofSignatureVerificationPatch.kt

There is no dependency declaration of microg. Aside that, do I have to make a PR? I think taking the time to reply here took more seconds than to actually implement my proposal.

**Reply by oSumAtrIX** [reply]
The dependency is from microg-support to that patch not the other way around. Remove it from the microg-support patch.

> Aside that, do I have to make a PR? I think taking the time to reply here took more seconds than to actually implement my proposal

No, it is certainly faster to respond to your it. Please open a PR, thanks.

**Reply by AgelosL** [reply]
See? Like I told you, that was neither hard nor slow to implement. After all, I decided to contribute my idea using the discussions forum and not a PR for a reason.

**Reply by oSumAtrIX** [reply]
No, you delayed it. I just happened to remember implementing it while working on something else. Especially if you think it's easy to do. This is an open source initiative not paid work.

**Reply by AgelosL** [reply]
_Why **discuss** it when you can make a **PR**?_
--------------------------------------------------------
Here are the facts. I made a suggestion using the channel I thought was more appropriate for me to contribute through - this discussion forum.
A suggestion which I think, we both agree, will make people happy and fix some serious problems since it's reported not only by me, but many other users as well. I did not know where the file that needed to be altered was hence this thread and not a **PR**.

Reasons for me **discussing** it and not making a **PR**:

Self-evident and pretty clear. If I knew how to **do** it, I wouldn't **discuss it**. Not only from a user's pov but also if I was the project's maintainer. I'd **do it** so I don't waste other peoples' **time** and mine. 

_What is a faster way to have the relative fix implemented,
deleting a single line or wasting other people's time by pretending to help and then complaining and "explaining" that open source is not paid work?_
--------------------------------------------------------

You replied by saying "go make a PR" which I tried doing. I tried following your unclear instructions that had **no link to the file needed editing** or even the **absolute minimum** of a **file's name** I should look into, and searched for "@dependson" in the patch I had aforementioned -signature spoofing- only to find that it's not there, since the file needing to be modified was the micro-g one, not the signature spoofing one. That lack of helpful info on your first comment shows how little you care about fixing something that worsens the users' experience. If you didn't care that much, you'd simply delete the line, not leave a second comment.

And the thing is that, in that second comment of yours, you fail again to admit the obvious and continue to waste my time, pretending to be dumb in order to avoid admitting that it's faster to delete a single line of code, knowing fully well of the location of the line, the filename etc. You are very fond of gaslighting and that shows.

So to sum up, the reasons why you chose to discuss and not delete a single line of code: I really don't know. There's something wrong with your attitude and it's going to affect the revanced project negatively by discouraging users to contribute ideas and developers working on it. This is my second suggestion and I chose to make this suggestion and not falter, even after you didn't correct the product banner wording thing. OxrxL, on the other hand, fed up at some point and stopped caring enough to contribute, after growing tired of trying to make valid points and I'm beginning to understand and see why he acted this way.

Lastly, I don't know what your opinion about this question is and I frankly couldn't care less about it, but for me, I don't think that in order to contribute in some way in an opensource project it's necessary for someone to know and have the project's structural architecture memorized. I do not know of **single** open source project that is improved **purely by pull requests**. Universally the norm is like any and all other experiences I had with anything that is open source: users are encouraged and welcomed to make suggestions through a forum like this.

So save any blame that you may have for me for not trying to make a PR for yourself. I can and I do blame you, for providing zero actual help, and topping it off by claiming I "delayed the fix". I can't delay a fix when I don't know how to make one. **That's why I suggested it by discussing it and not making a PR**. If someone **delayed the fix**, that is **you** by giving out _vague instructions_, and _assuming I know what what needs to be changed and where, and how to do it_.

Moral of the story:
--------------------------------------------------------

-Don't assume other people, while having the knowledge to create relevant PRs implementing fixes, instead go ahead and suggest the fix in the discussions forum.

Not everyone is like you. Most people, when something needs to be fixed, if they know what needs to be done, they go ahead and delete a single line of code, not give out vague and unfruitful instructions and purposefully gaslight in order to shed any real blame from themselves.

-I know this open source work is not paid work and I never acted like I paid you to do something you failed to provide on an imaginary deadline. 
Fwiw, my first comment on revanced's repos was this, on revanced-patches PR#641:

[https://github.com/revanced/revanced-patches/pull/641#issuecomment-1262624261](url)

> I don't want to sound like a judging dick about it, but I really am curious as to why hasn't this been merged yet.. Also, and I hope this goes without saying, I, and I hope the rest of the community appreciates your wonderful work you've put into this after vanced was discontinued.

Asking a simple question of what remains to be implemented and thanking you and OxrxL for your work. 

Therefore, please try and refrain from feeling attacked when someone suggests you implement a fix that would benefit countless users on an opensource project you are managing. If you don't care enough to strive for excellence for your project at least stop acting like you do. And certainly don't accuse people that have thanked you for your contributions of not appreciating you.

**Reply by oSumAtrIX** [reply]
> Why discuss it when you can make a PR?

No, I did not say you should open a PR instead of discussing it.  I said to open a PR after we discussed.

> If I knew how to do it, I wouldn't discuss it.

That is a very flawed approach. You should always discuss before opening a PR and getting it rejected due to lack of discussion of the implementation for example.

> and searched for "@dependsOn" in the patch I had aforementioned -signature spoofing-

Wrong, I clearly said to remove it from another patch as seen here: 
https://github.com/orgs/revanced/discussions/941#discussioncomment-5361493

> If you didn't care that much, you'd simply delete the line, not leave a second comment.

You incorrectly assume simplicity here. Changes have to be locally tested, PRed, reviewed and merged. You were asked to do the first two.

> order to avoid admitting that it's faster to delete a single line of code,

Once again you assume things incorrectly here. You need to locally test your changes. You generally don't work off the GitHub web editor for that reason as a convention. At that point in time I wrote the comment I was not available to test changes locally as I was not home nor had time.

> by discouraging users to contribute ideas 

That is incorrect, you were not discouraged to contribute ideas, you were encouraged to make a PR in addition to your suggestion.

>  I do not know of single open source project that is improved purely by pull requests.

Then you have seen a proper open source project. PRs are the main source and sometimes even the only source of adding code to a codebase

>  If someone delayed the fix, that is you by giving out vague instructions

Incorrect, I have given the clearest instructions possible at that point:

> by removing the @dependsOn annotation from the patch class file.
> Remove it from the microg-support patch.


TLDR: Dunning Kruger, you assume too much without understanding the reality. You delayed the fix by missreading what I clearly said, which is to remove the said annotation from the said class.

**Reply by AgelosL** [reply]
> No, I did not say you should open a PR instead of discussing it. I said to open a PR after we discussed.

> That is a very flawed approach. You should always discuss before opening a PR and getting it rejected due to lack of discussion of the implementation for example.

The one assuming too much is you. I didn't start to discuss so I could then make a PR. I only discussed a matter I thought needed some attention. The fact that you are still fixated on me making a PR proves that you haven't understood anything.

>Wrong, I clearly said to remove it from another patch as seen here:
https://github.com/orgs/revanced/discussions/941#discussioncomment-5361493

>remove the dependency of microg-support to that patch by removing the @dependsOn annotation from the patch class file

Again, you assumed that I knew the line that needed to be deleted was in the micro-g patch file, and that I know how revanced patches are made. Simple as that. There is no excuse for not using a link since I've seen you do that in other PRs.

> Once again you assume things incorrectly here. You need to locally test your changes. You generally don't work off the GitHub web editor for that reason as a convention. At that point in time I wrote the comment I was not available to test changes locally as I was not home nor had time.

I never said that you should implement my suggestion or merge any PR I would make without testing. Another baseless assumption on your part. Not to mention that I didn't even submit a PR for you to merge or anything like that.

> That is incorrect, you were not discouraged to contribute ideas, you were encouraged to make a PR in addition to your suggestion.

I am entitled to my opinion and I believe I know better than you on whether I am discouraged by your actions and general attitude or not. You discourage users from contributing their ideas when you assume that they are capable of making a PR a priori. Some users don't know how to code and their ideas shouldn't remain ideas when they can't be translated to PRs. Therefore, if I want my contribution to be in a form of an idea/suggestion and not a PR bc I simply can't produce one, I shouldn't refrain from making it. That is why I say you discourage people from contributing. And you also discourage people when you don't accept corrections on simple wording issues like "Hide banner to view products".

> Then you have seen a proper open source project. PRs are the main source and sometimes even the only source of adding code to a codebase

Again, assuming I want to contribute to the codebase. I never said so. I only said that including signature spoofing patch when including the micro-g patch creates problems for users that had none up until this point.  

> Incorrect, I have given the clearest instructions possible at that point:

You maybe think you were clear. You weren't.

> TLDR: Dunning Kruger, you assume too much without understanding the reality. You delayed the fix by missreading what I clearly said, which is to remove the said annotation from the said class.

AFAIK, the Dunning Kruger effect is when someone is overestimating his knowledge or ability of something. After explaining countless time that I didn't create a PR because of my lack of knowledge of how revanced patches work, I am surprised that you still believe you can seriously accuse me of something like that. 

If anything you are the one suffering from that Dunning Kruger effect, unable to stop assuming.

**Comment by oSumAtrIX** [comment]
> The one assuming too much is you

No that is incorrect. You assumed too much as shown in my previous comment.

> I didn't start to discuss so I could then make a PR

And? You generally discuss before opening a PR. Its routine. 

> The fact that you are still fixated on me making a PR proves that you haven't understood anything.

The fact is not a fact because the PR was already done, meaning i can't be fixated on you making a PR. So this is wrong as well.

> Again, you assumed that I knew the line that needed to be deleted was in the micro-g patch file

Correct. I assume you can locate a line of string given the name of the file and the line of string. If you can not, I advice you to learn how to use a computer. 

> There is no excuse for not using a link since I've seen you do that in other PRs.

I am not obliged to do anything in the first place. For me it was convenient at the moment to type the name of the patch in question. You are free to ask for further help for example if you don't know where or how to find it. Yet you refused to even bother. Chosers cant be beggars.

> I never said that you should implement my suggestion or merge any PR I would make without testing.Another baseless assumption on your part. 

Incorrect, I didn't say you said it. Therefor there is no assumption. It needs to be done is what I said and at that point I was not available on the PC. You are free to open a discussion, I am free to ask you to open a PR. You opened the discussion in the hope to get something fixed, you consider as an issue. Refusing to collab is a stab in your own foot.

> You discourage users from contributing their ideas when you assume that they are capable of making a PR a prior.

Incorrect. I have not requested an entire patch, I requested something which you are fully capable of doing but were entitled enough to refuse and ignore to contribute. I will remain asking for peoples contribution if I think it is reasonable. Anyone can ask for help in return or decline if they have their reasons or can't do it for example. You on the other hand stab on your own foot if you don't want to help me help you fix your issue.

> Some users don't know how to code and their ideas shouldn't remain ideas when they can't be translated to PRs

Correct, some users don't know how to code. As previously explained, if that applies to you, you could've simply said that uppon my request. Ideas will remain ideas if I want them to remain ideas in this organization. As a maintainer of the patches I am to chose, if something gets implemented or not, it is at no time your right to chose if it gets added or not. You are free to fork and implement it yourself (aka a PR with hoops). An idea remains a request I can either accept or decline with no further explanation required if i want. Of course I am not like that and consider every idea, reason why I review each idea, but I am not forced to implement an idea like you try to imply let it be any idea.

> Therefore, if I want my contribution to be in a form of an idea/suggestion and not a PR bc I simply can't produce one, I shouldn't refrain from making it

Correct. You are free to open a discussion without requiring to make a PR. I never declined to begin with. Yet I am also free to ask you to open a PR. If you don't do it (as your right) I am either free to pick it up or not. I am not obliged to as explained previously.

> That is why I say you discourage people from contributing

This is false. As explained previously any contribution is welcome. A discussion is also a form of contribution. I at no point discourage to discuss, like you assume wrongly again, even the opposite, I currently encourage to discuss before writing any code so that the written code is not wasted in the case of wrong impl. You opened a discussion, I suggested you to open a PR based in that discussion. If you were to open as discussed, the chances it gets implemented is high. You are free to ask any other team member, they will confirm that I encourage discussion before implementation a lot, proving your assumption wrong.

> And you also discourage people when you don't accept corrections on simple wording issues like "Hide banner to view products

Wrong. As a maintainer of the repository I will keep a certain standard in code quality and conventions. If someone suggests changes, I will review them and based on the review either dismiss the change or accept. In the example you gave, I discussed and explained the problem throughoul, leading to a dismissal of the code change. 

> Again, assuming I want to contribute to the codebase. I never said so

A discussion is a form of contribution as previously explained. It reveals a problem for example which is the first step towards improving something. Of course if you refuse to contribute further, you shoot yourself in the foot, as if you dont fix it, I might not as well and there wont be a solution to what you consider an issue for yourself. As simple as that.

>  I only said that including signature spoofing patch when including the micro-g patch creates problems for users that had none up until this point.

Correct, I evaluated this and suggested necessary changes to fix the issue. As we are open source you were free to PR discussed changes. 

> You maybe think you were clear. You weren't.

Incorrect. I stated what to change and where. This is cristal clearly stated in my previous comments. Of course, you might still not understand something at which you are free to tell me and ask for further explanation. You refused to do.

> AFAIK, the Dunning Kruger effect is when someone is overestimating his knowledge or ability of something

Correct, this is currently the case for you. Most things you claimed or assumed were either partially or completely incorrect.


> After explaining countless time that I didn't create a PR because of my lack of knowledge of how revanced patches work, I am surprised that you still believe you can seriously accuse me of something like that.

Incorrect, you have not said that you are incapable a single time before I made the PR. You simply didn't read my PR inquiry correctly on which i gave you a correction and explained what you understood wrong. For that you remain a prime example of Dunning Kruger.

**Reply by AgelosL** [reply]
I think my original suggestion was pretty clear from the very beginning. I simply stated that signature spoofing should be excluded when patching when micro-g patch was enabled, since it creates many problems and solves none. This simple suggestion differs completely from what you ridiculously accuse me of. 

You act as if I asked you to do the PR , implementing it immediately or stating that I wanted to make that PR myself and asking for your or anyone else's help implementing my idea. In reality I just brought attention to something I thought needed it, which is entirely different from asking an immediate implementation. Something you seem incapable of understanding and accepting.

What I did was make a **simple suggestion** to which you replied **I needed to submit a PR**. That was the moment you brought the word PR into the conversation. And that, because of your stupid assumption.

Now, instead of asking for a **PR** from me, you could have easily gone ahead and say that you are aware of the problems signature spoofing creates when patched along micro-g and that you agree with me or, that you don't agree with me because you regard these problems as non-trivial. That would both be fine for me either way and would have ended the discussion right then and there, but you didn't. 

Instead, you assumed that I either I demanded an immediate implementation by you or that I asked for your help in creating the implementation myself, both of which I didn't ask. So, save any accusations of begging of PRs or implementations or better yet, shove them up yours.

And all this, as if my suggestion was to benefit only me and not every other user that experiences the same problems as I do. That exposes 1) your inability to comprehend simple English language and 2) your habit of quick and unfounded assumptions based on the Dunning-Kruger effect you experience, which incidentally, is the only thing you are equipped to adequately understand and describe.

But you don't stop there.

> do I have to make a PR?

>I decided to contribute my idea using the discussions forum and not a PR for a reason.

Here we are, 24 hours later and you still blame me for a lack of a PR even after later acknowledging that a PR is not the only way of contributing in an OS project. Which brings me to my next question:

On the one hand you say you requested a simple line deletion, not an entire patch completely ignoring my quotes:

> I have not requested an entire patch, **I requested something which you are fully capable of doing** but were entitled enough to refuse and ignore to contribute

and later you follow that with:

> Correct. You are free to open a discussion **without requiring to make a PR**. 

So which is it? Is just my original suggestion enough to be considered a meaningful way of contributing or the contribution can only be in the form of a PR? Do I have to stop after explaining my suggestion and the logic behind it for it to be considered a contribution or do I have to create a PR? 

-------------------------------

> I am not obliged to do anything in the first place. For me it was convenient at the moment to type the name of the patch in question. You are free to ask for further help for example if you don't know where or how to find it. Yet you refused to even bother. Chosers cant be beggars.

> Refusing to collab is a stab in your own foot. 

> You on the other hand stab on your own foot if you don't want to help me help you fix your issue

> Of course, if you refuse to contribute further, you shoot yourself in the foot, as if you don’t fix it, I might not as well and there won’t be a solution to what you consider an issue for yourself. As simple as that.

By repeating the same strawman argument, blaming me that I "don't want to help", "shoot myself in the foot", and being a choosing beggar even though I tried helping by making the PR you requested, are the fanciest words your 1 cell structure of an excuse you call brain could hope to think of for a poor excuse to say you are not discouraging other people to contribute with just their ideas.

A **simple suggestion** is not a plea or begging for **PR**, it's just that, a **suggestion**. If you can't understand this simple sentence go seek help from a professional english language teacher. 

And all this crap about me "feeling entitled enough to refuse to contribute", all the talk about me "shooting myself in the foot" and being a choosing beggar, especially since they come from a person with the mental capacity of an amoeba, one that can't comprehend simple English language and speaks with alternating single line quotes and turds as excuses for sentences, is complete and utter shit. The only fault I can and will accept for myself is that, I unintentionally gave you permission to think that you and your arguments have any real value by exercising optimism that somehow at some point, you will understand. 

But by trying to make you understand simple concepts, I ended up wasting my time, and that's why I will end this discussion here. I am tired of your endless sealioning, strawmaning and gaslighting and general logical fallacy extravaganza you got going on here. Again, best course of action is to seek help from a professional language teacher if you fail this hard to grasp these kind of easy lingual concepts.

Sorry it took so much time to make you self-conscious of your inabilities to discuss. But that's not what I am here for. I only suggested something that would benefit almost everyone. At the end If you want your users to be frustrated and have a shitty experience using your open source project, go ahead and never read the discussions or make excuses for not changing something yourself. OxRxL made the best available move taking a dump on this and you and I'm sure inotia will follow soon.

So much for "remember implementing it while working on something else" when in fact, another person gave you more reasons why signature spoofing should be disabled.

https://github.com/revanced/revanced-patches/pull/1755#issuecomment-1475320867
